email_alert_service/models/email_alert.rb

```diff
@@ -43,4 +44,24 @@ def email_api_client
   def strip_empty_arrays(tag_hash)
     tag_hash.reject { |_, tags| tags.empty? }
   end
+
+  def document_links
+    strip_empty_arrays(
+      document.fetch("links", {}).merge("taxon_tree" => taxon_tree)
+    )
+  end
+
+  def taxon_tree
+    return [] unless document.dig("links", "taxons")
+
+    # TODO: update the public API of the taxonomy helpers gem to also accept a
+    # single fully expanded document so that we don't have to pass in the
+    # document twice.
+    linked_content_item = GovukTaxonomyHelpers::LinkedContentItem.from_publishing_api(
+      content_item: document,
+      expanded_links: document,
```

Comment:
Instead of making a per-representation factory method, did we consider exposing the primitive methods that flattens the structure?

This could be something like:

```ruby
GovukTaxonomyHelpers.taxons_with_ancestors(document.dig("expanded_links", "taxons"))
```

It feels confusing to pass in the `document` into the class, while we're not intending to use it.